experimental/packages/opentelemetry-instrumentation-http/src/http.ts

```diff
@@ -352,6 +351,12 @@ export class HttpInstrumentation extends InstrumentationBase<Http> {
       this._closeHttpSpan(span);
     });
 
+    request.on('timeout', (error: Err) => {
+      this._diag.debug('outgoingRequest on request timeout()', error);
+      utils.setSpanWithError(span, new Error('Timeout'));
+      this._closeHttpSpan(span);
```

Comment:
I'm not convinced that we should close the span on the timeout event. The [timeout event](https://nodejs.org/api/http.html#event-timeout) is purely an indication that the socket is inactive for a period of time. It doesn't close the request automatically. The span should be closed when the request is actually finished or errored.

Reply:
Added this as I found the span is not closed somehow in the integration test, thanks for pointing me to the right direction. Added in the `httpRequest.get` function to `emit('errro')` when timeout. Please check again.